app/api/v1/endpoints/strategies.py

```
@@ -263,23 +280,29 @@ async def execute_strategy(
             credits_used = credits_required
             
-            # Record credit transaction
+            # Record credit transaction with correct model fields
             credit_tx = CreditTransaction(
-                user_id=current_user.id,
+                account_id=credit_account.id,
                 amount=-credits_required,
-                transaction_type="strategy_execution",
-                description=f"Strategy: {request.function} on {request.symbol}",
-                reference_id=execution_result.get("execution_id")
+                transaction_type=CreditTransactionType.USAGE,
```

Ruff (0.12.2)

Undefined name `CreditTransactionType`

(F821)